server/uptime-calculator.js

```diff
@@ -741,21 +741,25 @@ class UptimeCalculator {
     }
 
     /**
-     * Get the uptime data by duration
-     * @param {'24h'|'30d'|'1y'} duration Only accept 24h, 30d, 1y
+     * Get the uptime data for given duration.
+     * @param {string} duration  A string with a number and a unit (h, d, or y), such as 24h, 30d, 1y.
      * @returns {UptimeDataResult} UptimeDataResult
      * @throws {Error} Invalid duration
      */
     getDataByDuration(duration) {
-        if (duration === "24h") {
-            return this.get24Hour();
-        } else if (duration === "30d") {
-            return this.get30Day();
-        } else if (duration === "1y") {
-            return this.get1Year();
-        } else {
+        const unit = duration.substring(duration.length - 1);
```

Comment:
You are make assumptions about the structure of the data passed to this function. Please return a more actionable error message to the user instead of bailing with integer parsing/.. errors.